pkg/staking/stakingcontract/contract.go

```diff
@@ -105,11 +104,13 @@ func (s *contract) sendDepositStakeTransaction(ctx context.Context, owner common
 }
 
 func (s *contract) getStake(ctx context.Context, overlay swarm.Address) (*big.Int, error) {
-	callData, err := stakingABI.Pack("stakeOfOverlay", overlay)
+	//overlayAddr is the byte array of fixed size and is required for smart contract.
+	var overlayAddr [32]byte
+	copy(overlayAddr[:], overlay.Bytes())
+	callData, err := stakingABI.Pack("stakeOfOverlay", overlayAddr)
```

Comment:
can we use `common.BytesToHash` ? Check how batch ID is handled in postage contract. Hash is [32]byte